hlapipeline/alignimages.py

```diff
@@ -3,65 +3,173 @@
-    imglist = input_list
+    print("-------------------- STEP 2: Filter data --------------------")
+    filteredTable = filter.analyze_data(imglist)
+
+    # Check the table to determine if there is any viable data to be aligned.  The
+    # 'doProcess' column (bool) indicates the image/file should or should not be used
+    # for alignment purposes.
+    if filteredTable['doProcess'].sum() == 0:
+        print("No viable images in filtered table - no processing done.\n")
+        return
+
+    # Get the list of all "good" files to use for the alignment
+    processList = filteredTable['imageName'][np.where(filteredTable['doProcess'])]
+    processList = list(processList) #Convert processList from numpy list to regular python list
+    print("\nSUCCESS")
 
     # 3: Build WCS for full set of input observations
-    refwcs = amutils.build_reference_wcs(imglist)
+    print("-------------------- STEP 3: Build WCS --------------------")
+    refwcs = amutils.build_reference_wcs(processList)
+    print("\nSUCCESS")
 
     # 4: Retrieve list of astrometric sources from database
-    reference_catalog = generate_astrometric_catalog(imglist, catalog='GAIADR2')
-
+    print("-------------------- STEP 4: Detect astrometric sources --------------------")
+    # While loop to accommodate using multiple catalogs
+    doneFitting = False
+    catalogIndex = 0
+    while not doneFitting:
+        reference_catalog = generate_astrometric_catalog(processList, catalog='GAIADR2', existing_wcs=refwcs)
+
+        # The table must have at least MIN_CATALOG_THRESHOLD entries to be useful
+        if len(reference_catalog) < MIN_CATALOG_THRESHOLD:
+            print("Not enough sources found in catalog" + catalogList[catalogIndex])
+            catalogIndex += 1
+            if catalogIndex <= (numCatalogs - 1):
+                print("Try again with the next catalog")
+                continue
+            else:
+                print("Not enough sources found in any catalog - no processing done.")
+                return
+        else:
+            doneFitting = True
+    print("\nSUCCESS")
     # 5: Extract catalog of observable sources from each input image
-    extracted_sources = generate_source_catalogs(imglist, refwcs, threshold=1000)
-
+    print("-------------------- STEP 5: Source finding --------------------")
+    extracted_sources = generate_source_catalogs(processList, refwcs, threshold=1000)
+    for imgname in extracted_sources.keys():
+        # The catalog of observable sources must have at least MIN_OBSERVABLE_THRESHOLD entries to be useful
+        # TODO *** Is this no hope or iterate on threshold or ???
+        if len(extracted_sources[imgname]["catalog_table"]) < MIN_OBSERVABLE_THRESHOLD:
+            print("Not enough sources ({}) found in image {}".format(len(extracted_sources[imgname]["catalog_table"]),imgname))
+            return
+
```

Comment:
Our original design discussion did not address directly the issue of not finding enough sources in the actual image if the image is part of a larger set of images.  I would think we should not return if the number of images is greater than one, but rather proceed as TweakReg does now.  It just notes that the image which does not satisfy the MIN_OBSERVABLE_THRESHOLD will not be used/aligned. The processList should be updated with the `doProcess` for this image which does not satisfy the "number of sources in image" criterion set to False.

Reply:
If there are no images which satisfy the criteria, we can return for now.  Once we have the bare bones of processing working, this would be a place to "twiddle some knobs" in experimentation (e.g., threshold?) to see if we can improve the situation.

Reply:
Issue to be addressed in merge of PR #37